src/sign_workflow/sign_existing_artifacts.py

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python
+
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+from pathlib import Path
+
+from sign_workflow.signer import Signer
+
+
+class SignExistingArtifacts:
+
+    def __init__(self, artifact_path, signature_type):
+        self.artifact_path = artifact_path
+        self.signature_type = signature_type
+
+    def __setattr__(self, name, value):
+        if name == 'artifact_path' and not isinstance(value, Path):
+            raise TypeError('SignExistingArtifacts.artifact_path must of type Path')
+        super().__setattr__(name, value)
+
+    def sign_using_manifest(self):
```

Comment:
did you mean `sign_existing_artifacts` here? using manifest already exists in https://github.com/opensearch-project/opensearch-build/pull/1352/files#diff-4bf5b6c3689571f89b11c463c1a6a4ff6bcdceb0a5e0ed1726d23886cb39ea18R24

Reply:
No, It is `SignExistingArtifacts` as in class name.

Reply:
why both methods are named same?

Reply:
fixed